benchmarks/nemo/nemolite2d/manual_versions/psykal_c_opencl/time_step_mod.f90

```diff
@@ -30,9 +30,10 @@ subroutine wrapper_c_invoke_time_step( &
                 sshn_t, sshn_u, sshn_v, hu, hv, un, vn, ua, ht, ssha_u, va, &
                 ssha_v, area_t, area_u, area_v, dx_u, dx_v, dx_t, dy_u, dy_v, &
                 dy_t, gphiu, gphiv
-            integer(c_intptr_t), intent(inout) :: ssha_t_dp, sshn_t_dp, &
+            type(c_ptr), intent(inout) :: ssha_t_dp, sshn_t_dp, &
                 sshn_u_dp, sshn_v_dp, hu_dp, hv_dp, un_dp, vn_dp, ua_dp, &
-                ht_dp, ssha_u_dp, va_dp, ssha_v_dp, tmask_dp, area_t_dp, &
+                ht_dp, ssha_u_dp, va_dp, ssha_v_dp
+            integer(c_intptr_t), intent(inout) :: tmask_dp, area_t_dp, &
```

Comment:
The dl_esm_inf update only affected the fields.

The grid properties are still just an integer handler (without any infrastructure to read/write) in dl_esm_inf. This is subject to https://github.com/stfc/dl_esm_inf/issues/18 and I agree they ultimately need to be just a `type(c_ptr)` to avoid any implementation assumption.

I can probably target that issue next/soon, but at the moment dl_esm_inf still expects c_intptr_t

Reply:
I finally decided to create a new dl_esm_inf PR, so this can be fixed in this here.